sbom-utility-scripts/scripts/index-image-sbom-script/index_image_sbom_script.py

```diff
@@ -55,26 +55,15 @@ def digest_hex_val(self) -> str:
         _, val = self.digest.split(":")
         return val
 
-    def purls(self, index_digest: Optional[str] = None) -> list[str]:
-        ans = []
-        if index_digest and self.arch:
-            ans.append(
-                PackageURL(
-                    type="oci",
-                    name=self.name,
-                    version=index_digest,
-                    qualifiers={"arch": self.arch, "repository_url": self.repository},
-                ).to_string()
-            )
-        ans.append(
-            PackageURL(
-                type="oci",
-                name=self.name,
-                version=self.digest,
-                qualifiers={"repository_url": self.repository},
-            ).to_string()
-        )
-        return ans
+    def purls(self, index_digest: str) -> list[str]:
+        purl = PackageURL(
+            type="oci",
+            name=self.name,
+            version=index_digest,
+            qualifiers={"arch": self.arch, "repository_url": self.repository},
+        ).to_string()
+
+        return [purl]
 
     def propose_spdx_id(self) -> str:
         purl_hex_digest = hashlib.sha256(self.purls()[0].encode()).hexdigest()
```

Comment:
We're missing an argument for `self.purls()` here.

I think the script deserves a small refactor to better reflect what it does now. Previously, we would create image objects with different `arch` and `digest` and pass the `image_index_digest` to the purls method to get more purls. We don't really want to do that anymore - IMO all the `Image` objects we create should have the same digest (the `image_index_digest`) and differ only in `arch`

Reply:
And the `purls()` method should take no parameters, it should just use `self.digest`